### DISCUSSION TITLE
Allow to add RPC method to AnyRpc server

### DISCUSSION THREAD

**Discussion by fdgStilla** [faaxm/spix#84] [General]
It would be great to be able to add some custom rpc methods in AnyRpcServer.
However I did not find a solution without exposing anyrpc in the header of AnyRpcServer like this `void addMethod(anyrpc::Method* method);`

But by looking at the name of this class, does it really matter to hide anyrcp from AnyRpcServer ?

**Comment by faaxm** [comment]
Hi @fdgStilla,
Spix has a "generic command" for which you can set a handler using `TestServer::setGenericCommandHandler`. Then, from your test script, you can use the `command` command to invoke the handler.
Would this already help with what you are trying to do?

**Reply by fdgStilla** [reply]
Yes this is a good start, thanks. But we have to write our own serialization and parsing, when we could use the one from AnyRpc.
